src/server/router.ts

```diff
@@ -0,0 +1,212 @@
+import { TRPCError } from "@trpc/server";
+import { procedure, router } from "./trpc";
+import { previewBuildConfigSchema, siteConfigSchema } from "../index";
```

Comment:
Hi! Unhelpfully, I haven't reviewed anything other than this. But, heads up: If you import this, the function builder needs to trace the entire dependency tree in `../index.ts` and you may end up pulling in the entire SDK extension dependency tree into your endpoint bundle. You'll definitely get faster bundle times and possibly a smaller bundle if you move these schemas to a different file.